pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -313,15 +313,13 @@ synchronized void runIncrementalPipeline()
   }
 
   /**
-   * This method periodically checks on Thrift server output directory and commands to create
-   * resource tables. The very first run happens over a minute later once controller spins up. And
-   * it's programmed to check on Thrift server every hour in case it went down for unfortunate
-   * reasons.
+   * This method checks upon Controller start checks on Thrift sever to create resource tables if
+   * they don't exist.
    *
    * @throws IOException
    * @throws SQLException
    */
-  @Scheduled(initialDelay = 60000, fixedDelay = 3600000)
+  @PostConstruct
```

Comment:
We already have a `@PostConstruct` method, please consider merging this with that (e.g., call this method at the end). From the [documentation](https://docs.oracle.com/javaee/7/api/javax/annotation/PostConstruct.html) it seems only one method should be annotated as such.

Reply:
Ok, so I did try merging both the PostConstruct into single method and I stumbled upon a strange issue: dataProperties.getThriftserverHiveConfig() was resorting to null and was throwing NPE. I could not figure out reason and now I have implemented application listener which will listen on ApplicationReadyEvent.